src/XMakeBuildEngine/UnitTests/Definition/ToolsetConfigurationReader_Tests.cs

```diff
@@ -488,6 +500,182 @@ public void PropertyTest_GetValueByName()
 
         #endregion
         #endregion
+
+        #region Extensions Paths
+        /// <summary>
+        ///  Tests multiple extensions paths from the config file, specified for multiple OSes
+        /// </summary>
+        [Fact]
+        public void ExtensionPathsTest_Basic1()
+        {
+            // NOTE: for some reason, <configSections> MUST be the first element under <configuration>
+            // for the API to read it. The docs don't make this clear.
+
+            ToolsetConfigurationReaderTestHelper.WriteConfigFile(ObjectModelHelpers.CleanupFileContents(@"
+                 <configuration>
+                   <configSections>
+                     <section name=""msbuildToolsets"" type=""Microsoft.Build.Evaluation.ToolsetConfigurationSection, Microsoft.Build"" />
+                   </configSections>
+                   <msbuildToolsets default=""2.0"">
+                     <toolset toolsVersion=""2.0"">
+                       <property name=""MSBuildBinPath"" value=""D:\windows\Microsoft.NET\Framework\v2.0.x86ret\""/>
+                       <property name=""MSBuildToolsPath"" value=""D:\windows\Microsoft.NET\Framework\v2.0.x86ret\""/>
+                       <msbuildExtensionsPathSearchPaths>
+                         <searchPaths os=""windows"">
+                            <property name=""MSBuildExtensionsPath"" value=""c:\foo""/>
+                            <property name=""MSBuildExtensionsPath64"" value=""c:\foo64;c:\bar64""/>
+                         </searchPaths>
+                         <searchPaths os=""osx"">
+                            <property name=""MSBuildExtensionsPath"" value=""/tmp/foo""/>
+                            <property name=""MSBuildExtensionsPath32"" value=""/tmp/foo32;/tmp/bar32""/>
+                         </searchPaths>
+                         <searchPaths os=""unix"">
+                            <property name=""MSBuildExtensionsPath"" value=""/tmp/bar""/>
+                         </searchPaths>
+                       </msbuildExtensionsPathSearchPaths>
+                     </toolset>
+                   </msbuildToolsets>
+                 </configuration>"));
+
+            Configuration config = ToolsetConfigurationReaderTestHelper.ReadApplicationConfigurationTest();
+            ToolsetConfigurationSection msbuildToolsetSection = config.GetSection(s_msbuildToolsets) as ToolsetConfigurationSection;
+
+            Assert.Equal(msbuildToolsetSection.Default, "2.0");
+            Assert.Equal(1, msbuildToolsetSection.Toolsets.Count);
```

Comment:
Oops :( Sure, I can fix it after this is merged. Was there anything else wrong with the file?